src/tileset/wangset.rs

```diff
@@ -0,0 +1,101 @@
+use std::collections::HashMap;
+
+use xml::attribute::OwnedAttribute;
+
+use crate::{
+    error::Error,
+    properties::{parse_properties, Properties},
+    util::{get_attrs, parse_tag, XmlEventResult},
+    Result, TileId,
+};
+
+mod wang_color;
+pub use wang_color::WangColor;
+mod wang_tile;
+pub use wang_tile::{WangId, WangTile};
+
+/// Undocummented WangSet types
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum WangType {
+    Corner,
+    Edge,
+    Mixed,
+}
+
+impl Default for WangType {
+    fn default() -> Self {
+        WangType::Mixed
+    }
+}
+
+/// Raw data belonging to a WangSet.
+#[derive(Debug, PartialEq, Clone)]
+pub struct WangSet {
+    /// The name of the Wang set
+    pub name: String,
+    /// Type of wangset
+    pub wang_type: WangType,
+    /// The tile ID of the tile representing this Wang set.
+    pub tile: Option<TileId>,
```

Comment:
Assuming this is correct, but just to be sure; @bjorn This is referring to a local ID in the tileset and not a GID right?

Reply:
Yes, since Wang sets are stored as part of the tileset, these tile references use the local IDs.